docker-app/qfieldcloud/core/admin.py

```diff
@@ -634,11 +634,16 @@ def lookups(self, request, model_admin):
         )
 
     def queryset(self, request, queryset):
-        q = Q(status="pending") | Q(status="started") | Q(status="queued")
-        if self.value() == "not finalized":
-            return queryset.filter(q)
-        else:
-            return queryset.filter(~q)
+        not_finalized = Q(last_status="pending") | Q(last_status="started")
+        match self.value():
+            case "ALL":
+                return queryset
+            case "not finalized":
+                return queryset.filter(not_finalized)
+            case "finalized":
+                return queryset.filter(~not_finalized)
+            case _:
+                messages.warning(request, f"Status {self.value()} is not implemented!")
```

Comment:
Do not use `match/case` keywords in the qfieldcloud codebase. They are unintuitive for people coming from other languages and may lead to errors I would prefer we never open this door. Python survived for 30+ years with simple `if/else` and it is among most popular languages.

When the default "All" option is selected in the UI, the `self.value()` returns `None`, not the string `ALL`. In the initial comment in the original PR/commit I never explicitly said this is the returned value in such cases.

Finally, in the `case _:`, it should `assert` or raise an exception, because such situation means the developer forgot to add the option in the `lookups`.

Reply:
Link to original discussion: https://github.com/opengisch/qfieldcloud/pull/545#discussion_r1139986268

Reply:
something like this should do it. I have not tested myself, some github python writing.

```
    def queryset(self, request, queryset):
        value = self.value()

        if value is None:
            return queryset

        q = Q(status="pending") | Q(status="started") | Q(status="queued")          
        if value == "not finalized":
            return queryset.filter(q)
        elif value == "finilized":
            return queryset.filter(~q)
        else:
            raise NotImplementedError(f"Uknown filter {value=}")
```